arviz/data/io_pyro.py

```diff
@@ -0,0 +1,89 @@
+"""pyro-specific conversion code."""
+import numpy as np
+
+from .inference_data import InferenceData
+from .base import dict_to_dataset
+
+
+def _get_var_names(posterior):
+    """Extract latent and observed variable names from pyro.MCMC.
+
+    Parameters
+    ----------
+    posterior : pyro.MCMC
+        Fitted MCMC object from Pyro
+
+    Returns
+    -------
+    list[str], list[str]
+    observed and latent variable names from the MCMC trace.
+    """
+    sample_point = posterior.exec_traces[0]
+    nodes = [node for node in sample_point.nodes.values() if node['type'] == 'sample']
+    observed = [node['name'] for node in nodes if node['is_observed']]
+    latent = [node['name'] for node in nodes if not node['is_observed']]
+    return observed, latent
+
+
+class PyroConverter:
+    """Encapsulate Pyro specific logic."""
+
+    def __init__(self, posterior, *_, coords=None, dims=None):
+        """Convert emcee data into an InferenceData object.
+
+        Parameters
+        ----------
+        posterior : pyro.MCMC
+            Fitted MCMC object from Pyro
+        coords : dict[str] -> list[str]
+            Map of dimensions to coordinates
+        dims : dict[str] -> list[str]
+            Map variable names to their coordinates
+        """
+        self.posterior = posterior
+        self.observed_vars, self.latent_vars = _get_var_names(posterior)
+        self.coords = coords
+        self.dims = dims
+
+    def posterior_to_xarray(self):
+        """Convert the posterior to an xarray dataset."""
+        from pyro.infer import EmpiricalMarginal
```

Comment:
Why does the import have to be here rather than at the top?

Reply:
Not making pyro a library requirement (the `io_pymc` does a similar trick)